apis/datadoghq/v2alpha1/datadogagent_default.go

```diff
@@ -0,0 +1,362 @@
+	defaultLogContainerLogsPath          string = "/var/lib/docker/containers"
+	defaultLogPodLogsPath                string = "/var/log/pods"
+	defaultLogContainerSymlinksPath      string = "/var/log/containers"
+	defaultLogTempStoragePath            string = "/var/lib/datadog-agent/logs"
+
+	defaultLiveProcessCollectionEnabled   bool = false
+	defaultLiveContainerCollectionEnabled bool = true
+
+	defaultOOMKillEnabled        bool = true
+	defaultTCPQueueLengthEnabled bool = false
+
+	defaultAPMEnabled         bool   = false
+	defaultAPMHostPortEnabled bool   = false
+	defaultAPMHostPort        int32  = 8126
+	defaultAPMSocketEnabled   bool   = true
+	defaultAPMSocketPath      string = "/var/run/datadog/apm.sock"
+
+	defaultCSPMEnabled              bool = false
+	defaultCWSEnabled               bool = false
+	defaultCWSSyscallMonitorEnabled bool = false
+
+	defaultNPMEnabled         bool = false
+	defaultNPMEnableConntrack bool = true
+	defaultNPMCollectDNSStats bool = true
+
+	defaultUSMEnabled bool = false
+
+	defaultDogStatsDOriginDetectionEnabled bool   = false
+	defaultDogStatsDHostPortEnabled        bool   = false
+	defaultDogStatsDPort                   int32  = 8125
+	defaultDogStatsDUseSocketVolume        bool   = false
+	defaultDogStatsDSocketPath             string = "/var/run/datadog/dsd.socket"
+
+	defaultCollectKubernetesEvents bool = true
+
+	defaultAdmissionControllerEnabled          bool = false
+	defaultAdmissionControllerMutateUnlabelled bool = false
+
+	defaultOrchestratorExplorerEnabled bool   = true
+	DefaultOrchestratorExplorerConf    string = "orchestrator-explorer-config"
+
+	defaultExternalMetricsServerEnabled bool = false
+	defaultDatadogMetricsEnabled        bool = true
+	// Cluster Agent versions < 1.20 should use 443
+	defaultMetricsProviderPort int32 = 8443
+
+	defaultKubeStateMetricsCoreEnabled bool   = false
+	defaultKubeStateMetricsCoreConf    string = "kube-state-metrics-core-config"
+
+	defaultClusterChecksEnabled    bool = true
+	defaultUseClusterChecksRunners bool = true
+
+	defaultPrometheusScrapeEnabled                bool = false
+	defaultPrometheusScrapeEnableServiceEndpoints bool = false
+
+	defaultDatadogMonitorEnabled bool = false
+)
+
+// DefaultDatadogAgent defaults the DatadogAgentSpec GlobalConfig and Features.
+func DefaultDatadogAgent(dda *DatadogAgent) {
+	defaultGlobalConfig(&dda.Spec)
+
+	defaultFeaturesConfig(&dda.Spec)
+}
+
+// defaultGlobalConfig sets default values in DatadogAgentSpec.Global.
+func defaultGlobalConfig(ddaSpec *DatadogAgentSpec) {
+	if ddaSpec.Global.ClusterName == nil {
+		defaultClusterNameCopy := defaultClusterName
+		ddaSpec.Global.ClusterName = &defaultClusterNameCopy
+	}
+
+	defaultCredentials(ddaSpec)
+}
+
+// defaultCredentials sets default values Credentials in the DatadogAgentSpec
+func defaultCredentials(ddaSpec *DatadogAgentSpec) {
+	if _, ok := ddaSpec.Override[ClusterAgentComponentName]; ok && *ddaSpec.Override[ClusterAgentComponentName].Disabled {
```

Comment:
I'm not sure this works in all cases. For instance if Cluster Agent is not explicitly disabled but no activated feature requires the Cluster Agent?

Also, I believe it will continuously change the `Token`, especially if we don't store the `defaulted` part in the `status`, no? I am not sure it should be in the defaulting part.
Maybe it's more something that should be handled in the code directly when checking if there's already a Secret and so on.